### PR TITLE
First pass at multi-language test matrix

### PR DESCRIPTION
**Pull request by sectioneight**
I think this could be cleaner if we put a ``Makefile`` in each
subdirectory, or maybe just a single top-level Makefile that runs all
the tests, but this is a start.

### CODE REVIEW

**Comment by kriskowal** [comment]
Good idea.

**Comment by jcorbin** [comment]
:+1: on having a Makefile for this, especially:
- as we add more languages
- as we add cross-language integration testing, we'll need some demarcation point for "run a client, run a server, ..."

If you're down for it, let's start with something like the following:
```
# node/Makefile
PHONY: install test

install:
  npm install

test:
  npm test
```

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,15 @@
 ---
 language: python
 
+env:
+  - SUBDIR=node
```
Comment (sectioneight) [inline .travis.yml]: I have no idea what node version this is using. What versions do we want to target for CI? In python, we're using `tox` to do multi-language (2.7 and 3.4 currently)

Comment (jcorbin) [inline .travis.yml]: There's not a great story for solving node version skew afaik, we baked our own internally to solve this problem. +@raynos for feedback on how this is usually dealt with wrt travis.

Comment (Raynos) [inline .travis.yml]: just configure travis to be 0.10

File: node/Makefile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.PHONY: install test_ci
+
+install:
+	npm install
+
+test_ci:
```
Comment (Raynos) [inline node/Makefile]: can we also add `test:` that runs `npm test`.

If we start a `cd folder && make test` pattern for local vs `cd folder && make test_ci` pattern for CI that might be nice.

Comment (jcorbin) [inline node/Makefile]: yeah can we just have:
```
test_ci: test

test:
  npm test
```

since there's no difference for node

Comment (sectioneight) [inline node/Makefile]: I did it the other way around, but same effect.

File: node/Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+.DEFAULT_GOAL := test
+
+install:
+	npm install
+
+test_ci:
+	npm test
+
+test: test_ci
```
Comment (sectioneight) [inline node/Makefile]: like this

**Comment by jcorbin** [comment]
:+1: merge at will

File: .travis.yml (hunk 1 above)
Comment (Raynos) [inline .travis.yml]: Can we just try setting

```
node_js:
  - "0.10"
  - "0.12"
```

and see what it dose ?

**Comment by Raynos** [comment]
This PR looks good to ship to me.

We can try iterating on multiple node languages in a seperate PR ?

**Comment by sectioneight** [comment]
Yea let's experiment with that next.

**Pull request merged**